virtwho/virt/virt.py

```diff
@@ -301,23 +301,42 @@ def __init__(self, logger, config, source=None, dest=None,
         super(IntervalThread, self).__init__()
 
     def wait(self, wait_time):
-        '''
+        """
         Wait `wait_time` seconds, could be interrupted by setting _terminate_event or _internal_terminate_event.
-        '''
+        """
         for i in range(wait_time):
             if self.is_terminated():
                 break
             time.sleep(1)
 
     def is_terminated(self):
         """
-
         @return: Returns true if either the internal terminate event is set or
                  the terminate event given in the init is set
         """
         return self._internal_terminate_event.is_set() or \
             self.terminate_event.is_set()
 
+    def is_registered(self):
+        """
+        This function tries to guess if system is registered according last attempt of
+        sending data to candlepin server.
+        :return: Return true, when virt-who was able to send data to candlepin server.
+        """
+
+        try:
+            is_registered = self.dest.get(key='is_registered', default=True)
```

Comment:
Using the existing datastore that is intended for reports allows for a sort of strange case to appear.
If a user has defined a configuration section with the name '[is_registered]' that value will end up being the value used here.

I've done this just to see what the behaviour is. For me the result is the destination thread refuses to send anything.

The config I used was as follows:

```ini
[is_registered]
type=libvirt
server=qemu:///system
owner=admin
env=admin
rhsm_hostname=http://localhost
rhsm_port=8443
rhsm_username=admin
rhsm_password=admin
rhsm_prefix=/candlepin
```

I think that storing system state in a shared (locked) dictionary is a good idea but perhaps it should be a different dictionary. Perhaps restrict the user from naming sections whatever values we'd like to store in the shared dict?

The other issue that these changes causes is when there are multiple destination threads (less common but expected). When that is the case in theory they'd all be trying to set the same value 'is_registered'.